definitions/mw.lua

```diff
@@ -0,0 +1,650 @@
+---@param text string
+---@param length number
+---@param ellipsis string?
+---@param adjustLength boolean
+---@return string
+function mw.text.truncate(text, length, ellipsis, adjustLength) end
+
+---Replaces MediaWiki <nowiki> strip markers with the corresponding text. Other types of strip markers are not changed.
+---@param s string
+---@return string
+function mw.text.unstripNoWiki(s) end
+
+---Equivalent to mw.text.killMarkers( mw.text.unstripNoWiki( s ) ).
+---@param s string
+---@return string
+function mw.text.unstrip(s) end
+
+
+---@class Title
+---@field id number
+---@field interwiki string
+---@field namespace number
+---@field nsText string
+---@field subjectNsText string
+---@field text string
+---@field prefixedText string
+---@field fullText string
+---@field rootText string
+---@field baseText string
+---@field subpageText string
+---@field canTalk string
+---@field exists boolean
+---@field file File
+---@field fileExists boolean
+---@field isContentPage boolean
+---@field isExternal boolean
+---@field isLocal boolean
+---@field isRedirect boolean
+---@field isSpecialPage boolean
+---@field isSubpage boolean
+---@field isTalkPage boolean
+---@field contentModel string
+---@field basePageTitle Title
+---@field rootPageTitle Title
+---@field talkPageTitle Title?
+---@field subjectPageTitle Title
+---@field redirectTarget Title|false
+---@field protectionLevels table
+---@field cascadingProtection table
+mw.title = {}
+
+---@class File
+---@field exists boolean
+---@field width number
+---@field height number
+---@field pages {width: number, height: number}[]?
+---@field size number
+---@field mimeType string
+---@field length number
+
+---Test for whether two titles are equal. Note that fragments are ignored in the comparison.
+---@param a Title
+---@param b Title
+---@return boolean
+function mw.title.equals(a, b) end
+
+---Returns -1, 0, or 1 to indicate whether the title a is less than, equal to, or greater than title b.
+---@param a Title
+---@param b Title
+---@return -1|0|1
+function mw.title.compare(a, b) end
+
+---Returns the title object for the current page.
+---@return Title
+function mw.title.getCurrentTitle() end
+
+---Creates a new title object. This function is expensive when called with an ID.
+---@param text string
+---@param namespace string?
+---@return Title?
+---@overload fun(id: number):Title?
+function mw.title.new(text, namespace) end
+
+---Creates a title object with title title in namespace namespace, optionally with the specified fragment and interwiki prefix. namespace may be any key found in mw.site.namespaces. If the resulting title is not valid, returns nil.
+---@param namespace string
+---@param title string
+---@param fragment string?
+---@param interwiki string?
+---@return Title?
+function mw.title.makeTitle(namespace, title, fragment, interwiki) end
```

Comment:
maybe note that `mw.title.makeTitle` unlike `mw.title.new` will always apply the specified namespace.